blocks/library/embed/index.js

```diff
@@ -73,52 +72,130 @@ registerBlock( 'core/embed', {
 		}
 	},
 
-	edit( { attributes, setAttributes, focus, setFocus } ) {
-		const { url, title, caption } = attributes;
+	edit: class extends wp.element.Component {
+		constructor() {
+			super( ...arguments );
+			this.doServerSideRender = this.doServerSideRender.bind( this );
+			this.state = {
+				html: '',
+				type: '',
+				error: false,
+				fetching: false,
+			};
+			this.noPreview = [
+				'facebook.com',
+			];
+			if ( this.props.attributes.url ) {
+				// if the url is already there, we're loading a saved block, so we need to render
+				this.doServerSideRender();
+			}
+		}
 
-		if ( ! url ) {
-			return (
-				<Placeholder icon="cloud" label={ wp.i18n.__( 'Embed URL' ) } className="blocks-embed">
-					<input type="url" className="placeholder__input" placeholder={ wp.i18n.__( 'Enter URL to embed here...' ) } />
-					<Button isLarge>
-						{ wp.i18n.__( 'Embed' ) }
-					</Button>
-				</Placeholder>
+		componentWillUnmount() {
+			// can't abort the fetch promise, so let it know we will unmount
+			this.unmounting = true;
+		}
+
+		doServerSideRender( event ) {
+			if ( event ) {
+				event.preventDefault();
+			}
+			const { url } = this.props.attributes;
+			const api_url = wpApiSettings.root + 'oembed/1.0/proxy?url=' + encodeURIComponent( url ) + '&_wpnonce=' + wpApiSettings.nonce;
+
+			this.setState( { error: false, fetching: true } );
```

Comment:
Maybe. I'd have to look again at where this function is called originally, and if by `componentDidMount`, why the component would not be mounted at this point.

To a more general point, I think this is a handy reference for when side effects and state changes should occur during component lifecycle:

https://gist.github.com/bvaughn/923dffb2cd9504ee440791fade8db5f9